cmd/tikv-ctl/src/main.rs

```diff
@@ -822,41 +822,60 @@ fn flashback_whole_cluster(
-                    let addr = pd_client.get_store(store_id).unwrap().address;
-                    leaders
-                        .into_iter()
-                        .filter(|(_, region_id)| {
-                            region_ids.is_empty() || region_ids.contains(region_id)
-                        })
-                        .for_each(|(key_range, region_id)| {
-                            // Prepare flashback region version by key range.
-                            let key_range = build_key_range(&key_range.0, &key_range.1, false);
-
-                            let addr = addr.clone();
-                            let cfg_inner = cfg.clone();
-                            let mgr = Arc::clone(&mgr);
-                            let f = async move {
-                                let debug_executor =
-                                    new_debug_executor(&cfg_inner, None, Some(&addr), mgr);
-                                debug_executor.flashback_to_version(
-                                    version,
-                                    region_id,
-                                    key_range,
-                                    start_ts.into_inner(),
-                                    0,
-                                )
-                            };
-                            futures.push(f);
-                        })
-                });
+            let read_result = prepare_stores_leader.clone();
+            read_result.into_iter().for_each(|(store_id, leaders)| {
+                let addr = pd_client.get_store(store_id).unwrap().address;
+                leaders
+                    .into_iter()
+                    .filter(|(_, region_id)| {
+                        region_ids.is_empty() || region_ids.contains(region_id)
+                    })
+                    .for_each(|(key_range, region_id)| {
+                        let key_range_var = key_range.clone();
+                        let mut prepare_stores_leader = prepare_stores_leader.clone();
+                        // Prepare flashback region version by key range.
+                        let key_range = build_key_range(&key_range.0, &key_range.1, false);
+
+                        let addr = addr.clone();
+                        let cfg_inner = cfg.clone();
+                        let mgr = Arc::clone(&mgr);
+                        let f = async move {
+                            let debug_executor =
+                                new_debug_executor(&cfg_inner, None, Some(&addr), mgr);
+                            match debug_executor.flashback_to_version(
+                                version,
+                                region_id,
+                                key_range.clone(),
+                                start_ts.into_inner(),
+                                0,
+                            ) {
+                                Ok(_) => {
+                                    prepare_stores_leader
```

Comment:
You just cloned `prepare_stores_leader` at line 843, so this seems meaningless since it won't affect the real `prepare_stores_leader` at all.

Reply:
fixed. now catch ref.